src/components/ContextMenu/ContextMenu.tsx

```diff
@@ -0,0 +1,155 @@
+import * as RightMenu from "@radix-ui/react-context-menu";
+import styled from "styled-components";
+import { Icon } from "..";
+import { Arrow, GenericMenuItem, GenericMenuPanel } from "../GenericMenu";
+import PopoverArrow from "../icons/PopoverArrow";
+
+export const ContextMenu = (props: RightMenu.ContextMenuProps) => (
+  <RightMenu.Root {...props} />
+);
+
+const ContextMenuTrigger = (props: RightMenu.ContextMenuTriggerProps) => {
+  return (
+    <RightMenu.Trigger
+      asChild
+      {...props}
+    />
+  );
+};
+
+ContextMenuTrigger.displayName = "ContextMenuTrigger";
+ContextMenu.Trigger = ContextMenuTrigger;
+
+const ContextMenuSubTrigger = ({
+  children,
+  ...props
+}: RightMenu.ContextMenuSubTriggerProps) => {
+  return (
+    <GenericMenuItem
+      as={RightMenu.SubTrigger}
+      {...props}
+    >
+      {children}
+      <div className="dropdown-arrow">
+        <Icon name="chevron-right" />
+      </div>
+    </GenericMenuItem>
+  );
+};
+
+ContextMenuSubTrigger.displayName = "ContextMenuSubTrigger";
+ContextMenu.SubTrigger = ContextMenuSubTrigger;
+
+export type ArrowProps = {
+  showArrow?: boolean;
+};
+
+type ContextMenuContentProps = RightMenu.MenuContentProps & {
+  sub?: true;
+} & ArrowProps;
+
+type ContextMenuSubContentProps = RightMenu.MenuSubContentProps & {
+  sub?: never;
+} & ArrowProps;
+
+const RightMenuContent = styled(GenericMenuPanel)<{ $showArrow?: boolean }>`
+  flex-direction: column;
+  z-index: 1;
```

Comment:
do we really need this z-index? z-index should be consider an escape hatch and very rarely used as it's very prone to create messy situations.

Reply:
This is added as there is an issue in storybook where the storybook code block has a higher z-index and there was a possibility of someone assigning such a high index so we added this
The code is to the end of the body but would still need a z-index 1 otherwise it takes auto and can get content over it
This would also be used for select inside a dialog